src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -13,67 +14,162 @@
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.server.ResponseStatusException;
 
-
+import java.sql.Date;
 import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @CrossOrigin(origins = "http://localhost:3000")
 public class AssignmentController {
-
+    @Autowired
+    SectionRepository sectionRepository;
+    @Autowired
+    AssignmentRepository assignmentRepository;
+    @Autowired
+    EnrollmentRepository enrollmentRepository;
+    @Autowired
+    GradeRepository gradeRepository;
 
     // instructor lists assignments for a section.  Assignments ordered by due date.
     // logged in user must be the instructor for the section
     @GetMapping("/sections/{secNo}/assignments")
-    public List<AssignmentDTO> getAssignments(
-            @PathVariable("secNo") int secNo) {
+    public List<AssignmentDTO> getAssignments(@PathVariable("secNo") int secNo) {
+        Section sec = sectionRepository.findById(secNo).orElse(null);
+
+        if (sec == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "section not found");
+        }
+
+        List<Assignment> assignments = assignmentRepository.findBySectionNoOrderByDueDate(secNo);
+        List<AssignmentDTO> assignmentsDTO = new ArrayList<>();
 
-        // TODO remove the following line when done
-		
-		// hint: use the assignment repository method 
-		//  findBySectionNoOrderByDueDate to return 
-		//  a list of assignments
+        assignments.forEach(ass -> assignmentsDTO.add(new AssignmentDTO(
+            ass.getAssignmentId(),
+            ass.getTitle(),
+            ass.getDueDate().toString(),
+            sec.getCourse().getCourseId(),
+            sec.getSecId(),
+            sec.getSectionNo()
+        )));
 
-        return null;
+        return assignmentsDTO;
     }
 
+    // TODO: trim title dto string to remove extra white spaces at start and end
     // add assignment
     // user must be instructor of the section
     // return AssignmentDTO with assignmentID generated by database
     @PostMapping("/assignments")
-    public AssignmentDTO createAssignment(
-            @RequestBody AssignmentDTO dto) {
+    public AssignmentDTO createAssignment(@RequestBody AssignmentDTO dto) {
+        Date dueDate = null;
+        try {
+            dueDate = Date.valueOf(dto.dueDate());
+        } catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
 
-        // TODO remove the following line when done
+        Section sec = sectionRepository.findById(dto.secNo()).orElse(null);
```

Comment:
Suggestion - Could simply code with the following and then additional checks would not be needed
```Section section = sectionRepository.findById(dto.secNo()).orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid section"));```